packages/core/src/core/config/env.spec.ts

```diff
@@ -99,37 +105,101 @@ describe('Env', () => {
       testConfigFile('.env.test', 'test');
     });
 
+    context('given NODE_ENV is defined and .env.${NODE_ENV}.local exists', () => {
+      testConfigFile('.env.test.local', 'test');
+    });
+
     context('given NODE_ENV is not defined and .env.development exists', () => {
       testConfigFile('.env.development');
     });
 
+    context('given NODE_ENV is not defined and .env.development.local exists', () => {
+      testConfigFile('.env.development.local');
+    });
+
     context('given a .env file exists', () => {
       testConfigFile('.env');
     });
 
+    context('given a .env.local file exists', () => {
+      testConfigFile('.env.local');
+    });
+
     context('given multiple config files exist', () => {
 
       afterEach(() => {
         removeFile('.env');
+        removeFile('.env.local');
         removeFile('.env.development');
+        removeFile('.env.development.local');
       });
 
       describe('should return the configuration value', () => {
 
-        it('with default.yml overriding default.js', () => {
-          writeFileSync('.env.development', dotEnvContent2);
+        it('and determine the variable values, based on configuration file importance', () => {
+          writeFileSync('.env.development.local', dotEnvContent4);
+          writeFileSync('.env.development', dotEnvContent3);
+          writeFileSync('.env.local', dotEnvContent2);
           writeFileSync('.env', dotEnvContent);
 
+          strictEqual(Env.get('FOO_BAR_FOUR'), 'four');
+          strictEqual(Env.get('FOO_BAR_THREE'), 'three');
+          strictEqual(Env.get('FOO_BAR'), 'hello2');
+        });
+
+        it('and should not delete configuration values (deep merge).', () => {
+          writeFileSync('.env.development.local', dotEnvContent4);
+          writeFileSync('.env.development', dotEnvContent3);
+          writeFileSync('.env.local', dotEnvContent2);
+          writeFileSync('.env', dotEnvContent);
+
+          strictEqual(Env.get('HELLO'), 'world');
+          strictEqual(Env.get('FOO_BAR_WITH_QUOTES_AND_WHITESPACES'), '  with whitespaces  ');
```

Comment:
Removing this line to better isolate the features tested by each `it` (otherwise this test will also test the `"` feature).